TSOClient/tso.simantics/NetPlay/EODs/Handlers/VMEODSlotsPlugin.cs

```diff
@@ -0,0 +1,678 @@
+
+            // atempt to credit the machine by debiting the owner
+            var VM = client.vm;
+            VM.GlobalLink.PerformTransaction(VM, false, client.Avatar.PersistID, Server.Object.PersistID, Convert.ToInt32(depositAmount),
+
+            (bool success, int transferAmount, uint uid1, uint budget1, uint uid2, uint budget2) =>
+            {
+                //TODO: Make this part of global link
+                VM.SendCommand(new VMNetAsyncResponseCmd(0, new VMTransferFundsState
+                {
+                    Responded = true,
+                    Success = success,
+                    TransferAmount = transferAmount,
+                    UID1 = uid1,
+                    Budget1 = budget1,
+                    UID2 = uid2,
+                    Budget2 = budget2
+                }));
+                if (success)
+                {
+                    client.Send("slots_resume_manage", "-" + amountString);
+                }
+                else
+                {
+                    client.Send("slots_deposit_fail", "" + amountString);
+                }
+            });
+        }
+        private void GameOverHandler(short eventID, VMEODClient player)
+        {
+            // check if object has enough money or too much money, then start new game or disconnect the player
+            if (IsObjectBalanceInBounds() == true)
+            {
+                player.Send("slots_new_game", "");
+                State = VMEODSlotsStates.Idle;
+            }
+            else
+            {
+                State = VMEODSlotsStates.Off;
+                userClient.Send("slots_toggle_offline_message", "");
+            }
+        }
+        private void BetHandler(string evt, string betAmountString, VMEODClient client)
+        {
+            short betAmount = Int16.Parse(betAmountString);
+            // attempt to debit player the bet amount
+            var VM = client.vm;
+            
+            VM.GlobalLink.PerformTransaction(VM, false, client.Avatar.PersistID, Server.Object.PersistID, Convert.ToInt32(betAmount),
+
+            (bool success, int transferAmount, uint uid1, uint budget1, uint uid2, uint budget2) =>
+            {
+                // add bet to the balance in object
+                machineBalance += betAmount;
+                //TODO: Make this part of global link
+                VM.SendCommand(new VMNetAsyncResponseCmd(0, new VMTransferFundsState
+                {
+                    Responded = true,
+                    Success = success,
+                    TransferAmount = transferAmount,
+                    UID1 = uid1,
+                    Budget1 = budget1,
+                    UID2 = uid2,
+                    Budget2 = budget2
+                }));
+                if (success)
+                {
+                    // roll the slots rng and get result, send to UI and send to SlotResultHandler() to determine win
+                    SlotResultHandler(RollNewGame(betAmount));
```

Comment:
This is running in the background thread (callback from global link). Modifying state from outwith the main loop of the VM could cause a race condition. The VM.SendCommand and userClient send functions are thread safe, everything else is up to you.